examples/use.hs

```diff
@@ -83,22 +82,15 @@ data StateMsg s res where
 
 
 state :: s -> Actor (StateMsg s) ()
-state s0 = do
-  stateIORef <- liftIO $ newIORef s0
-
-  forever $ receive \case
-    GetState ->
-      liftIO $ readIORef stateIORef
-
-    GetsState f -> do
-      s <- liftIO $ readIORef stateIORef
-      pure (f s)
-
-    PutState s ->
-      liftIO $ writeIORef stateIORef s
-
-    ModifyState f ->
-      liftIO $ modifyIORef stateIORef f
+state = loop
+  where
+    loop s = do
```

Comment:
Strictly speaking, there's no need for `loop`, the implementation could be inlined in `state` and tail-call `state` at the end, but I prefer to write the code as-is, not recursing a top-level function into itself, when possible.